src/color.c

```diff
@@ -607,6 +612,75 @@ static int parse_gnu_ls_colors(struct colors *colors, const char *ls_colors) {
 	return ret;
 }
 
+#if __FreeBSD__
+/** Parse the FreeBSD $LSCOLORS format. */
+static int parse_bsd_ls_colors(struct colors *colors, const char *lscolors) {
+	int ret = -1;
+	static const dchar *keys[LSCOLORS_KEYS] = {
+		"di", "ln", "so", "pi", "ex", "bd", "cd", "su", "sg", "tw", "ow"
+	};
+
+	static const char default_colors[LSCOLORS_LEN] = "exfxcxdxbxegedabagacad";
+
+	static const char *fg_codes[256] = {
+		['a'] = "30", ['b'] = "31", ['c'] = "32", ['d'] = "33",
+		['e'] = "34", ['f'] = "35", ['g'] = "36", ['h'] = "37", ['x'] = "39",
+		['A'] = "1;30", ['B'] = "1;31", ['C'] = "1;32", ['D'] = "1;33",
+		['E'] = "1;34", ['F'] = "1;35", ['G'] = "1;36", ['H'] = "1;37", ['X'] = "1"
+	};
+
+	static const char *bg_codes[256] = {
+		['a'] = "40", ['b'] = "41", ['c'] = "42", ['d'] = "43",
+		['e'] = "44", ['f'] = "45", ['g'] = "46", ['h'] = "47", ['x'] = "49",
+		['A'] = "100", ['B'] = "101", ['C'] = "102", ['D'] = "103",
+		['E'] = "104", ['F'] = "105", ['G'] = "106", ['H'] = "107", ['X'] = "49"
+	};
+
+	char complete_colors[LSCOLORS_LEN + 1];
+
+	if (!lscolors || !lscolors[0]) {
+		memcpy(complete_colors, default_colors, LSCOLORS_LEN);
+	} else {
+		int elems = strlen(lscolors);
+		if ((elems == 0) || ((elems % 2) != 0) || (elems > LSCOLORS_LEN)) {
+			goto exit;
```

Comment:
There's no cleanup, so you might as well just
```suggestion
			return -1;
```